modules/tfm/tfm/boards/common/plat_init.c

```diff
@@ -16,7 +16,11 @@
 enum tfm_hal_status_t tfm_hal_platform_init(void)
 {
 	__enable_irq();
+
+	/* Only if UART1 is used by TF-M do we initialize it. */
+	#ifdef CONFIG_SECURE_UART1
```

Comment:
This is a downstream platform init function overriding the TF-M weak implementation.

So why do we need this PR upstream ?
https://github.com/zephyrproject-rtos/zephyr/pull/38670

From what I can tell, disabling secure uart1 upstream will cause a bus fault on nordic boards because TF-M upstream in Zephyr will always execute: 
https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/c74be3890c9d975976fde1b1a3b2f5742bec34c0/trusted-firmware-m/platform/ext/common/tfm_platform.c#L16

I guess what we should have, was simply an extra setting here:
https://github.com/nrfconnect/sdk-nrf/blob/master/modules/trusted-firmware-m/Kconfig
allowing us to disable `SECURE_UART1` and then pass on that knowledge to TF-M build using:
```
# Set up configurations for TF-M on zephyr_property_target
  set_property(TARGET zephyr_property_target
    APPEND PROPERTY TFM_CMAKE_OPTIONS
      -DSECURE_UART1=OFF
  )
```
when secure uart1 is set to off.

This TF-M option is then automatically applied here:
https://github.com/zephyrproject-rtos/zephyr/blob/0343118abba34383def9032a1b934a14706005de/modules/trusted-firmware-m/CMakeLists.txt#L197
```
      $<GENEX_EVAL:$<TARGET_PROPERTY:zephyr_property_target,TFM_CMAKE_OPTIONS>>
```

Reply:
I believe that this is not exactly accurate.  We use either the platform initialization file in our dowstream TF-M: 

[https://github.com/nrfconnect/sdk-trusted-firmware-m/blob/master/trusted-firmware-m/platform/ext/common/tfm_platform.c](https://github.com/nrfconnect/sdk-trusted-firmware-m/blob/master/trusted-firmware-m/platform/ext/common/tfm_platform.c)

Or the file in here in sdk-nrf. I was not aware of the one in zephyr to be honest. 

In any case, adding a configuration option in our TF-M module here instead of doing that in zephyr maybe is a simpler solution and I cannot find any disadvantages. I will give this a try and update the PR accordingly.

Reply:
@tejlmand  @SebastianBoe  I updated the PR and closed the PRs in the downstream/upstream zehpyr.  This solution also seems to work ok. The default for the KConfig is yes here since we used to have it default on in our devices.

Reply:
> I believe that this is not exactly accurate. We use either the platform initialization file in our dowstream TF-M:

and our downstream TF-M is a fork of the Zephyr TF-M, not the upstream of the upstream TF-M:
![image](https://user-images.githubusercontent.com/9802583/134462899-02762e2c-2cf5-45a3-9966-5fd7b4e7500b.png)

Reply:
That is usefull to know, thanks!